src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -93,6 +93,45 @@ class CallCredentials(object):
   def __init__(self, low_credentials):
     self._low_credentials = low_credentials
 
+class GoogleOAuth2MetadataPlugin(interfaces.GRPCAuthMetadataPlugin):
```

Comment:
The functionality for both implementations of GRPCAuthMetadataPlugin is alright,
but I think we can't just expose these implementations in the public API like suggested here.

Once JWT functionality gets added to OAuth2 library, the jwt token will be obtained by just doing getting `self.credentials.get_access_token().access_token` (same as for OAuth2).

So we basically want a grpc specific  wrapper around  the "credentials" object (same functionality as GoogleOAuth2MetadataPlugin, but with a different name).

In terms of JWT, it is probably fine to have the class GoogleJWTAuthMetadataPlugin, but it shouldn't be exposed publicly (at least for now).